### DISCUSSION TITLE
Comparing structured textual data

### DISCUSSION THREAD

**Discussion by Gitoffthelawn** [WinMerge/winmerge#1551] [Q&A]
I have multiple text files, each with contents using the structure:

```
====================
Alpha:   12345
Beta:    12.395
Gamma:   1254.5
Delta:   1.2145
Epsilon: 5.6546
Zeta:    4.56874
Eta:     8787.90
====================

====================
Alpha:   66767.78
Beta:    67678
Gamma:   6.745689
Delta:   67.7678
Epsilon: 567.654
Zeta:    89467.786
Eta:     6783.9
====================

...
```

The rows of equal signs and the labels to the left of the colons are consistent.

The numbers are all that will vary between files.

Within any group (surrounded by a row of equal signs), if one of the values changes from one file to another, the entire group is considered different.

Each file has 500 groups.

Overall, WinMerge works very well to perform such comparisons.

The only snag is that, *between 2 files*, when one number is different within a group, often all the other numbers within the group are also different *between the files*.  Thus, when using WinMerge to review changes, I need to click the "Next Difference" button (or keyboard shortcut) 7 times just to get through a single group.

Is there a way to get WinMerge to recognize all text between 2 rows of equal signs as a single item, so it can be recognized as a single entity?

Ideally, I would like to be able to specify this functionality via the CLI.  If that's not possible, I could set up a unique file extension just for files of this structure.

**Comment by sdottaka** [comment]
Unfortunately, WinMerge does not allow such grouping.

One idea is to combine "Alpha: " to "Eta: " into one line.
If you specify the Unpacker plugin from the command prompt as shown below, it will look like the attached image.

"c:\Program Files\WinMerge\WinMergeU.exe" file1.txt file2.txt /unpacker "Replace -e (\d+)\r\n '$1, '|Replace = ''"

![image](https://user-images.githubusercontent.com/98126/199007080-788715f3-7c8d-4a02-8f23-aa69b97ad1df.png)
